pkg/ipamd/ipamd.go

```diff
@@ -391,10 +391,15 @@ func (c *IPAMContext) nodeInit() error {
 	}
 
 	primaryENIMac := c.awsClient.GetPrimaryENImac()
-	err = c.networkClient.SetupHostNetwork(vpcV4CIDRs, primaryENIMac, &primaryV4IP, c.enablePodENI, c.enableIPv4, c.enableIPv6)
+	ipt, err := c.networkClient.SetupHostNetwork(vpcV4CIDRs, primaryENIMac, &primaryV4IP, c.enablePodENI, c.enableIPv4, c.enableIPv6)
 	if err != nil {
 		return errors.Wrap(err, "ipamd init: failed to set up host network")
 	}
+	err = c.networkClient.CleanUpStaleAWSChains(ipt)
```

Comment:
I see all of the other `networkClient` functions that modify iptables rules calling `n.newIptables(ipProtocol)`. Why do we pass it as an argument here?

Reply:
I thought it would be better to pass down that variable. If it is unnecessary I can also call `n.newIptables(ipProtocol)` in the clean up function.